pep-0458.txt

```diff
@@ -280,140 +301,189 @@ from PyPI. TUF downloads them and checks them against the TUF metadata that it
 also downloads from the repository. If the downloaded target files are
 trustworthy, TUF then hands them over to the package manager.
 
-The `Metadata`__ document provides information about each of the required
-metadata and their expected content.  The next section covers the different
+The `Metadata`__ document provides information about each type of required
+metadata and its expected content.  The next section covers the different
 kinds of metadata RECOMMENDED for PyPI.
 
 __ https://github.com/theupdateframework/tuf/blob/v0.11.1/docs/METADATA.md
 
+In addition, all target files SHOULD be available on disk at least three times.
+Once under their original filename, to provide backwards compatibility, and
+twice with their SHA-256 and SHA-512 hash respectively included in their
```

Comment:
It feels to me like using both sha256 and sha512 here is needless extra cost. We don't know what kinds of attacks are going to be available in the future and which hash functions they're going to affect. I think It would be a better overall idea to have a single hashed location, plus the unhashed location. In the future if/when an attack against the current hash, we'll need a plan to transition to a new hash (which I think we need documented anyways, even with 2 hashes because the weakness could affect both of them).

Reply:
- Mostly addressed by removing sha256 in https://github.com/secure-systems-lab/peps/pull/71 and https://github.com/secure-systems-lab/peps/pull/74.
- Hash migration plan is still up for discussion (see https://github.com/secure-systems-lab/peps/issues/68#issuecomment-554283532). Maybe Python folks want to chime in?